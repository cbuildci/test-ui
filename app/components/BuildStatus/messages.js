/*
 * BuildStatus Messages
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
    buildStatus: {
        id: 'app.components.BuildStatus.buildStatus',
        defaultMessage: `{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }`,
    },
});
