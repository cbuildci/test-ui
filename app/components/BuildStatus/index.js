/**
 * BuildStatus
 */

import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import messages from './messages';

export const statusDisplay = {
    IN_PROGRESS: { icon: 'fas fa-hourglass-start', color: '#17a2b8' },
    WAITING_FOR_DEPENDENCY: { icon: 'far fa-clock', color: '#6c757d' },
    BUILD_NOTFOUND: { icon: 'fas fa-times', color: '#dc3545' },
    SUCCEEDED: { icon: 'fas fa-check', color: '#28a745' },
    DEPENDENCY_FAILED: { icon: 'fas fa-times', color: '#dc3545' },
    START_CODEBUILD_FAILED: { icon: 'fas fa-times', color: '#dc3545' },
    STARTING: { icon: 'fas fa-hourglass-start', color: '#17a2b8' },
    FAILED: { icon: 'fas fa-times', color: '#dc3545' },
    FAULT: { icon: 'fas fa-times', color: '#dc3545' },
    STOPPED: { icon: 'fas fa-stop-circle', color: '#dc3545' },
    TIMED_OUT: { icon: 'far fa-clock', color: '#dc3545' },
    SKIPPED: { icon: 'fas fa-share', color: '#6c757d' },
};

export function getStatusColor(status) {
    return (statusDisplay[status] || {}).color || null;
}

function BuildStatus({ status }) {
    const { color, icon } = statusDisplay[status] || {};

    return (
        <span style={{ color: color }}>
            {icon && <i className={`${icon} fa-fw`}/>}
            {icon && ' '}
            {status ? <FormattedMessage
                {...messages.buildStatus}
                values={{ status: status }}
            /> : '-'}
        </span>
    );
}

BuildStatus.propTypes = {
    status: PropTypes.string,
};

BuildStatus.defaultProps = {
    status: null,
};

export default BuildStatus;
