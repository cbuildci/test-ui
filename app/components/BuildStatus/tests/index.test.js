import React from 'react';
import { shallow } from 'enzyme';
import BuildStatus, { statusDisplay } from '../index';

describe('<BuildStatus />', () => {

    it('should have expected statuses for statusDisplay', () => {
        expect(Object.keys(statusDisplay).sort())
            .toEqual([
                'IN_PROGRESS',
                'WAITING_FOR_DEPENDENCY',
                'BUILD_NOTFOUND',
                'SUCCEEDED',
                'DEPENDENCY_FAILED',
                'START_CODEBUILD_FAILED',
                'STARTING',
                'FAILED',
                'FAULT',
                'STOPPED',
                'TIMED_OUT',
                'SKIPPED',
            ].sort());
    });

    it('should render expected JSX for statuses', () => {
        for (const status of Object.keys(statusDisplay)) {
            expect(shallow(
                <BuildStatus status={status}/>,
            )).toMatchSnapshot();
        }
    });

    it('should render expected JSX for unknown status', () => {
        expect(shallow(
            <BuildStatus status="foobar"/>,
        )).toMatchSnapshot();

        expect(shallow(
            <BuildStatus/>,
        )).toMatchSnapshot();
    });

});
