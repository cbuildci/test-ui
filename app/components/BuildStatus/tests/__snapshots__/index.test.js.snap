// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuildStatus /> should render expected JSX for status "" 1`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "" 2`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "BUILD_NOTFOUND" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "BUILD_NOTFOUND",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "BUILD_NOTFOUND" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Build Not Found
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "DEPENDENCY_FAILED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "DEPENDENCY_FAILED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "DEPENDENCY_FAILED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Dependency Failed
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "FAILED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "FAILED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "FAILED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Failed
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "FAULT" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "FAULT",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "FAULT" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Fault
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "IN_PROGRESS" 1`] = `
<span
  style={
    Object {
      "color": "#17a2b8",
    }
  }
>
  <i
    className="fas fa-hourglass-start fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "IN_PROGRESS",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "IN_PROGRESS" 2`] = `
<span
  style="color:#17a2b8"
>
  <i
    class="fas fa-hourglass-start fa-fw mr-1"
  />
  <span>
    In Progress
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "SKIPPED" 1`] = `
<span
  style={
    Object {
      "color": "#6c757d",
    }
  }
>
  <i
    className="fas fa-share fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "SKIPPED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "SKIPPED" 2`] = `
<span
  style="color:#6c757d"
>
  <i
    class="fas fa-share fa-fw mr-1"
  />
  <span>
    Skipped
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "START_CODEBUILD_FAILED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-times fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "START_CODEBUILD_FAILED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "START_CODEBUILD_FAILED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-times fa-fw mr-1"
  />
  <span>
    Failed to Start CodeBuild
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "STARTING" 1`] = `
<span
  style={
    Object {
      "color": "#17a2b8",
    }
  }
>
  <i
    className="fas fa-hourglass-start fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "STARTING",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "STARTING" 2`] = `
<span
  style="color:#17a2b8"
>
  <i
    class="fas fa-hourglass-start fa-fw mr-1"
  />
  <span>
    Starting CodeBuild
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "STOPPED" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="fas fa-stop-circle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "STOPPED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "STOPPED" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="fas fa-stop-circle fa-fw mr-1"
  />
  <span>
    Stopped
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "SUCCEEDED" 1`] = `
<span
  style={
    Object {
      "color": "#28a745",
    }
  }
>
  <i
    className="fas fa-check fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "SUCCEEDED",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "SUCCEEDED" 2`] = `
<span
  style="color:#28a745"
>
  <i
    class="fas fa-check fa-fw mr-1"
  />
  <span>
    Succeeded
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "TIMED_OUT" 1`] = `
<span
  style={
    Object {
      "color": "#dc3545",
    }
  }
>
  <i
    className="far fa-clock fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "TIMED_OUT",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "TIMED_OUT" 2`] = `
<span
  style="color:#dc3545"
>
  <i
    class="far fa-clock fa-fw mr-1"
  />
  <span>
    Timed Out
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "WAITING_FOR_DEPENDENCY" 1`] = `
<span
  style={
    Object {
      "color": "#6c757d",
    }
  }
>
  <i
    className="far fa-clock fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "WAITING_FOR_DEPENDENCY",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for status "WAITING_FOR_DEPENDENCY" 2`] = `
<span
  style="color:#6c757d"
>
  <i
    class="far fa-clock fa-fw mr-1"
  />
  <span>
    Waiting for Dependency
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for unknown status 1`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  <FormattedMessage
    defaultMessage="{status, select,
            IN_PROGRESS {In Progress}
            WAITING_FOR_DEPENDENCY {Waiting for Dependency}
            BUILD_NOTFOUND {Build Not Found}
            SUCCEEDED {Succeeded}
            DEPENDENCY_FAILED {Dependency Failed}
            START_CODEBUILD_FAILED {Failed to Start CodeBuild}
            STARTING {Starting CodeBuild}
            FAILED {Failed}
            FAULT {Fault}
            STOPPED {Stopped}
            TIMED_OUT {Timed Out}
            SKIPPED {Skipped}
            other {{status}}
        }"
    id="app.components.BuildStatus.buildStatus"
    values={
      Object {
        "status": "foobar",
      }
    }
  />
</span>
`;

exports[`<BuildStatus /> should render expected JSX for unknown status 2`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  <span>
    foobar
  </span>
</span>
`;

exports[`<BuildStatus /> should render expected JSX for unknown status 3`] = `
<span
  style={
    Object {
      "color": "#333333",
    }
  }
>
  <i
    className="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;

exports[`<BuildStatus /> should render expected JSX for unknown status 4`] = `
<span
  style="color:#333333"
>
  <i
    class="fas fa-question-circle fa-fw mr-1"
  />
  -
</span>
`;
